groups/ais/instance.tf

```diff
@@ -13,49 +13,65 @@ resource "aws_security_group" "common" {
   name   = "common-${local.common_resource_name}"
   vpc_id = data.aws_vpc.heritage.id
 
-  ingress {
-    description = "Allow SSH connectivity for application deployments"
-    from_port   = 22
-    to_port     = 22
-    protocol    = "TCP"
-    cidr_blocks = var.deployment_cidrs
-  }
+  tags = merge(local.common_tags, {
+    Name = "common-${local.common_resource_name}"
+  })
+}
 
-  dynamic "ingress" {
-    for_each = var.tuxedo_services
-    iterator = service
-    content {
-      description = "Allow OIS Tuxedo connectivity for Tuxedo ${upper(service.key)} services"
-      from_port   = service.value
-      to_port     = service.value
-      protocol    = "TCP"
-      cidr_blocks = data.aws_subnet.application[*].cidr_block
-    }
-  }
+resource "aws_security_group_rule" "ingress_ci_deployments" {
+  type              = "ingress"
+  description       = "Allow inbound SSH connectivity for CI application deployments"
+  from_port         = 22
+  to_port           = 22
+  protocol          = "TCP"
+  cidr_blocks       = var.deployment_cidrs
+  security_group_id = aws_security_group.common.id
+}
```

Comment:
Purely from a state indexing point of view, I would recommend making this a `for_each` loop, as follows:

```
resource "aws_security_group_rule" "ingress_ci_deployments" {
  for_each = toset(var.deployment_cidrs)

  type              = "ingress"
  description       = "Allow inbound SSH connectivity for CI application deployments"
  from_port         = 22
  to_port           = 22
  protocol          = "TCP"
  cidr_blocks       = [each.value]
  security_group_id = aws_security_group.common.id
}
```